bot/exts/events/advent_of_code/_cog.py

```diff
@@ -185,15 +186,47 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
         brief="Get a snapshot of the PyDis private AoC leaderboard",
     )
     @whitelist_override(channels=AOC_WHITELIST_RESTRICTED)
-    async def aoc_leaderboard(self, ctx: commands.Context) -> None:
-        """Get the current top scorers of the Python Discord Leaderboard."""
+    async def aoc_leaderboard(
+            self,
+            ctx: commands.Context,
+            dayandstar: Optional[str] = None,
+            maximum_scorers: Optional[int] = 10) -> None:
+        """
+        Get the current top scorers of the Python Discord Leaderboard.
+
+        Additionally, you can provide a day-star(eg.: 2-1, as second day first star) to filter your query more.
+        """
+        if maximum_scorers > AocConfig.max_dayandstar_results or maximum_scorers <= 0:
+            raise commands.BadArgument(
+                f"Unfortunately, \n the maximum number of results you can query is {AocConfig.max_dayandstar_results} "
+                "\n and the minimum number is 1"
+            )
```

Comment:
I don't think the newlines are necessary here, and the minimum intuitively makes sense in this context so it doesn't need to be stated.

```suggestion
            raise commands.BadArgument(
                f"Unfortunately, the maximum number of results you can query is {AocConfig.max_dayandstar_results}"
            )
```